optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/IntervalClusterImpl.java

```diff
@@ -102,58 +100,11 @@ public void addInterval(Interval<Interval_, Point_> interval) {
         count++;
     }
 
-    public Iterable<IntervalClusterImpl<Interval_, Point_, Difference_>>
-            removeInterval(Interval<Interval_, Point_> interval) {
-        // TODO: Make this incremental by only checking between the interval's
-        //       start and end points
-        return () -> new Iterator<IntervalClusterImpl<Interval_, Point_, Difference_>>() {
-
-            IntervalSplitPoint<Interval_, Point_> current = getStart(startSplitPoint);
-
-            private IntervalSplitPoint<Interval_, Point_>
-                    getStart(IntervalSplitPoint<Interval_, Point_> start) {
-                while (start != null && start.isEmpty()) {
-                    start = splitPointSet.higher(start);
-                }
-                return start;
-            }
-
-            @Override
-            public boolean hasNext() {
-                return current != null && current.compareTo(endSplitPoint) <= 0 && !splitPointSet.isEmpty();
-            }
-
-            @Override
-            public IntervalClusterImpl<Interval_, Point_, Difference_> next() {
-                IntervalSplitPoint<Interval_, Point_> start = current;
-                IntervalSplitPoint<Interval_, Point_> end;
-                int activeIntervals = 0;
-                count = 0;
-                boolean anyOverlap = false;
-                do {
-                    count += current.intervalsStartingAtSplitPointSet.size();
-                    activeIntervals +=
-                            current.intervalsStartingAtSplitPointSet.size() - current.intervalsEndingAtSplitPointSet.size();
-                    if (activeIntervals > 1) {
-                        anyOverlap = true;
-                    }
-                    current = splitPointSet.higher(current);
-                } while (activeIntervals > 0 && current != null);
-                hasOverlap = anyOverlap;
-
-                if (current != null) {
-                    end = splitPointSet.lower(current);
-                    current = getStart(current);
-                } else {
-                    end = splitPointSet.last();
-                }
-
-                return new IntervalClusterImpl<>(splitPointSet, differenceFunction, start, end, count, hasOverlap);
-            }
-        };
+    Iterable<IntervalClusterImpl<Interval_, Point_, Difference_>> removeInterval(Interval<Interval_, Point_> interval) {
```

Comment:
Interesting, the `interval` argument is unused.

Reply:
The entire cluster is recomputed since it not easy to get active interval count given a start point (It doesn't recompute all the datapoints, only the datapoints in the same cluster).